plugins/server/graphql-kotlin-graalvm-metadata-generator/src/main/kotlin/com/expediagroup/graphql/plugin/graalvm/GenerateGraalVmMetadata.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2023 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.plugin.graalvm
+
+import com.expediagroup.graphql.generator.SchemaGeneratorConfig
+import com.expediagroup.graphql.generator.TopLevelObject
+import com.expediagroup.graphql.generator.federation.FederatedSchemaGeneratorConfig
+import com.expediagroup.graphql.generator.federation.FederatedSchemaGeneratorHooks
+import com.expediagroup.graphql.generator.federation.toFederatedSchema
+import com.expediagroup.graphql.generator.hooks.NoopSchemaGeneratorHooks
+import com.expediagroup.graphql.generator.toSchema
+import com.expediagroup.graphql.plugin.graalvm.DefaultMetadataLoader.loadDefaultReflectMetadata
+import com.expediagroup.graphql.plugin.schema.hooks.SchemaGeneratorHooksProvider
+import com.expediagroup.graphql.server.Schema
+import com.expediagroup.graphql.server.operations.Mutation
+import com.expediagroup.graphql.server.operations.Query
+import com.expediagroup.graphql.server.operations.Subscription
+import io.github.classgraph.ClassGraph
+import io.github.classgraph.ScanResult
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.util.ServiceLoader
+
+private val logger: Logger = LoggerFactory.getLogger("generateGraalVmMetadata")
+
+/**
+ * Generate GraalVM reflect metadata for the underlying GraphQL schema.
+ */
+fun generateGraalVmMetadata(supportedPackages: List<String>): List<ClassMetadata> {
+    val defaultMetadata: List<ClassMetadata> = loadDefaultReflectMetadata()
+    val reflectMetadata: List<ClassMetadata> = generateGraalVmReflectMetadata(supportedPackages)
+    return reflectMetadata + defaultMetadata
+}
+
+/**
+ * Generate GraalVM reflect metadata for the underlying GraphQL schema.
+ */
+fun generateGraalVmReflectMetadata(supportedPackages: List<String>): List<ClassMetadata> {
```

Comment:
this should be `internal` function  but due to some Gradle (fixable) and IDE (open issue) issues it makes it simpler to just keep it public for now (module should generally be used from plugins anyway).

Reply:
updated tests so it can be internal